### ISSUE TITLE
Post Types: Allow organizers to override speaker avatars

### ISSUE THREAD

**Issue by ryelle** [WordPress/wordcamp.org#180]
Anecdotally this has come up a few times - speakers either don't have a gravatar account, or want to use a different headshot for the WordCamp site. Currently, organizers don't have a good way of overriding that.

One option could be enabling the featured image, but themes might use that on the page in general, causing awkward just unexpected layouts, where maybe the old gravatar is on the page along with the new featured image.

Another idea could be a new custom field to upload an image, which would ensure there's no theme compat issues, but a bit more work up front.

**Comment by iandunn** [comment]
Personally I'd consider this a `wontfix`. I can see how it's a minor inconvenience, but I think we can make a much bigger impact focusing our limited time on other things.

**Comment by ryelle** [comment]
I think this happens more than you think (see our [two mystery avatars on this WCUS announcement](https://2019.us.wordcamp.org/2019/08/01/announcing-our-speakers-round-4/)), and especially as conf sites are using larger photos (like in the speaker blocks, where it can go up to 600px), photos that are "designed" for 100px comment avatars are not necessarily best for speaker profile photos.

While I'm sure it's not a high priority, I wouldn't classify it `wontfix`.

**Comment by iandunn** [comment]
>mystery avatars

What's the reason that they don't just create Gravatars like everybody else?

IIRC, the only reason I've heard in the past was b/c the speaker had an over-inflated ego and didn't think it was important to do the things the organizer team asked them to do. IMO that's a huge red flag that they're probably not going to be a good speaker (in the holistic sense).

There could definitely be other reasons that I just haven't heard before, though.

> photos that are "designed" for 100px comment avatars are not necessarily best for speaker profile photos.

I'm not sure what you mean here, in my experience people upload the full photo to Gravatar, and you can just add `s=1600` (or whatever) to the URL to get an unpixelated photo at whatever size you need (up to the full size, of course). That's working fine in the speaker block as far as I can tell. I'm sure I'm missing something here, but don't see what it is.

**Comment by NewYorkerLaura** [comment]
This affects me a lot. It takes up considerable volunteer hours to explain to folks how to do this.  This doesn't count the hours I spend saying your Gravatar isn't connect to the email you gave me and they can't figure out what email it's connected to. And are annoyed that I can't override this. Some of these folks are CEO's of businesses that contribute major sums to the program, or are involved in Open Source in another capacity and are speaking on a topic like security or diversity that crosses over many areas.

There are plenty of folks who speak at WordCamps and are active in the WordPress Community via their local meetup who don't have gravatars because they are not active in the forums and don't contribute (which is a problem for another day on how a large chunk of communities in the US have no idea what contributing means or think it is coding folks only. The vast majority of USA based WordCamps have no contributor day. ). 

They also may want one photo that's more in keeping with a public speaking venture (WordCamps) vs something that represents them in a forum among folks they contribute with. In other words their headshot vs. a manga image. 

Additionally since Gravatrs are changeable whenever, someone could have an appropriate Gravatar one day and in less than appropriate one another. Since the WordCamp sites are always up you might have a speaker highlighted whose image isn't in keeping with the event. This year was the guy with a monkey sitting on him at Gibraltar peeing on him. 

On a PR and Marketing level it reflects poorly on the conference that folks have photos missing, and can't select something for that conference. It also just not what conferences do. I think WordCamps have evolved over time. Small local shindigs have grown, and we need to grow with the times. 

Back in 1996 when I first became active on the web everyone had a screen name and some sort of graphic that represented them. Until around 2008 or so, this was still commonplace, but the web has evolved. I still am Pelirroja_ljc in some old time forums and god only knows what avatar is attached to that name, but I'd never sign into a conference today with that moniker. 

The short version is that I think we need to evolve and bring ourselves to different level.

**Comment by iandunn** [comment]
This reminds me of a similar discussion we had several years ago, about [whether or not we should require speakers to create WordPress.org accounts](https://make.wordpress.org/community/2017/03/02/requiring-wordcamp-speakers-to-have-a-wordpress-org-account/). The result of that is that we do require them in order to complete CFP forms.

>  It takes up considerable volunteer hours to explain to folks how to do this. 
> This doesn't count the hours I spend saying your Gravatar isn't connect to the email you gave me and they can't figure out what email it's connected to. 

It seems like that could mitigated by linking to documentation. If Gravatar's existing docs aren't good enough, we could point out the places where they could be improved. I'm happy to find someone at Gravatar who could review any suggestions we have.

> Some of these folks are CEO's of businesses that contribute major sums to the program 

I don't see why they'd get special treatment because they're a CEO or give money. If they feel like they deserve that, that would be a big red flag IMO, because that feels like a toxic attitude to me.

If they're really that "important", then I'd assume that they could just as easily ask their assistant to do it for them.

> or are involved in Open Source in another capacity and are speaking on a topic like security or diversity that crosses over many areas.

I think that's a fair point, people from outside our community can lend a very important perspective, and it's very helpful to include them in camps.

I still don't see why they wouldn't be willing to spend 2 minutes setting up a simple Gravatar account, though. That doesn't feel like a big ask when we're giving them this kind of platform, and if they're not reliable enough to do something small, then that feels like a red flag that they'll be unreliable when it comes to big things like completing the slides, showing up on time, etc.


> There are plenty of folks who speak at WordCamps and are active in the WordPress Community via their local meetup who don't have gravatars because they are not active in the forums and don't contribute 

I feel similar to the above here, it's a simple ask when we're giving them a big platform and allowing them to represent WP.


> They also may want one photo that's more in keeping with a public speaking venture (WordCamps) vs something that represents them in a forum among folks they contribute with. In other words their headshot vs. a manga image.

IMO a headshot would be the most appropriate thing in forums too. Having a manga photo instead of a real one doesn't help to build a sense of community. People can do it if they want, but I don't think we should encourage it.


> Additionally since Gravatrs are changeable whenever, someone could have an appropriate Gravatar one day and in less than appropriate one another. 
> Since the WordCamp sites are always up you might have a speaker highlighted whose image isn't in keeping with the event. This year was the guy with a monkey sitting on him at Gibraltar peeing on him.

Why would we accept someone like that as a speaker in the first place? If that's how they already act in public, can we really trust them to be appropriate in their slides and on stage? 

If it happens after the event, we could just remove their speaker post. That seems like a rare occurance, though, and I'm always leery of enacting policies or spending time on outliers like that.

> On a PR and Marketing level it reflects poorly on the conference that folks have photos missing, 

I agree, but I think the appropriate solution is for those speakers to just do the very simple, quick, and easy task of setting up a Gravatar account. I still don't understand why that's so hard for this small group of people.

An alternative solution, though, might be to make it easier to upload a photo to Gravatar when editing your w.org profile. That would help all w.org users, and streamline the process for speakers outside of our community, since we already require them to create a w.org account.

> It also just not what conferences do. I think WordCamps have evolved over time. Small local shindigs have grown, and we need to grow with the times.

I **strongly** disagree with that; I think the heart of WordCamps are that they're _casual_ and _volunteer-organized_, we should *not* try to look like "professional" industry conferences IMO.

----

Having said all that, this is something I can disagree-and-commit on if the team wants to move forward.

It feels like it ventures beyond technical implementation and into values and priorities, though, so I think it'd be helpful to first get some feedback from @andreamiddleton and [maybe the wider community team as well](https://make.wordpress.org/community/feature-requests-for-community-sites/).

**Comment by NewYorkerLaura** [comment]
I hear what you are saying, but I strong disagree with what your perception is of how CEOs and their staff functions. I base this on decades of interaction with this type of clientele in the non-profit world.  Believe me, I don't like it, but it is what it is. 

I think you'd find your words ironic if you knew which CEO  we perceptually wait weeks and months to receive information from that gridlocks us, in 2018 the same CEO did the inviting of these folks that we struggled for weeks to to get basic info from, one of these guest couldn't get the time of their talk correct and created chaos the day of the event as three people were utterly derailed in what we were doing to find them and get them on stage, but will bring this to other places as you suggested. 

We are constantly being asked to grow this event (specifically WCUS not local camps) to a wider audience, to bring us the recognition of this event to a larger scale, to bring an awareness to people who have never been involved or heard about us before. It's not so much this one thing as it is a a thousand little things like this that combined make it more difficult to do what is being asked.  That take time from people who are volunteering what hours they have. On a marketing level to grow this program and respect volunteer hours things need to change. It doesn't operate as a local WordCamp on any level.

**Comment by ryelle** [comment]
Also reported on trac: https://meta.trac.wordpress.org/ticket/4734

**Comment by DeveloperWil** [comment]
I understand the point of representing WordPress on the stage but Gravatar has nothing to do with WordPress. It is used by WordPress but it's not a core or a WordPress service. Asking a speaker to create a new account on a service they may never use ever again just because we can't update a picture on a WordCamp site makes us sound ridiculous and reflects badly on us. Those are my thoughts and I'd like to be able to override gravatar images in speaker blocks too

**Comment by mundschenk-at** [comment]
Also Gravatar comes with a lot of privacy issues (for site visitors as well es users).

**Comment by bethsoderberg** [comment]
Commenting to say that I found this post because I wanted to know if I _could_ override the Gravatars because right now I have 17 out of 61 speakers (28%!) who don't have them. Speaker wrangling is already a huge job and I can attest to the fact that Gravatar wrangling takes up a lot of organizer/volunteer time. I disagree with relegating this to the "won't fix" pile as a fix for this  would save literally hundreds of volunteer hours (maybe more!) each year.

**Comment by francescamarano** [comment]
> What's the reason that they don't just create Gravatars like everybody else?

Because not everyone wants to register for a Gravatar :) 
Since the function is not part of WP Core or WordCamp functionality, I think we should find a different solution that doesn't require people to signup for third party services.

**Comment by NewYorkerLaura** [comment]
I feel like part of this is a bit of a the old[ Zen Cat Story. :](http://truecenterpublishing.com/zenstory/ritualcat.html)

**"When the spiritual teacher and his disciples began their evening meditation, the cat who lived in the monastery made such noise that it distracted them. So the teacher ordered that the cat be tied up during the evening practice.** 

**Years later, when the teacher died, the cat continued to be tied up during the meditation session. And when the cat eventually died, another cat was brought to the monastery and tied up. Centuries later, learned descendants of the spiritual teacher wrote scholarly treatises about the religious significance of tying up a cat for meditation practice."**

In the early days of WordPress, online culture throughout the web was very different. Social media wasn't prevalent. Online interaction were very forum driven with screen names instead of real ones and avatars instead of real photos. For a bunch of forums (EZBoards and Invision Boards anyone?) you had default options that came with the forum or you could upload your own by hot linking to the URL where they resided. So I get it, in yesteryear. Gravatar served a purpose. But online culture has changed, privacy has changed. Let's not keep tying the cat up because we always tied the cat up!

**Comment by iandunn** [comment]
😆, I like that story too. My favorite version of it is from [Company](https://www.indiebound.org/book/9781400079377).

<img width="464" alt="Screen Shot 2020-02-27 at 5 10 10 PM" src="https://user-images.githubusercontent.com/484068/75501174-128ce480-5984-11ea-9bd0-ed58b4299f06.png">

I don't think that's what's going on here, but if you'd like to propose it, I think posting on the Community p2 would be the next step. GitHub is good for discussing technical implementation, but I think proposal discussions should include the entire team.

---

If the team does want to do this, then I think we should do it in a way that doesn't add more work for organizers. For example, requiring potential speakers to upload a photo with their application.
